lib/buffer.js

```diff
@@ -323,30 +324,8 @@ Buffer.compare = function compare(a, b) {
 };
 
 
-Buffer.isEncoding = function(encoding) {
-  var loweredCase = false;
-  for (;;) {
-    switch (encoding) {
-      case 'hex':
-      case 'utf8':
-      case 'utf-8':
-      case 'ascii':
-      case 'latin1':
-      case 'binary':
-      case 'base64':
-      case 'ucs2':
-      case 'ucs-2':
-      case 'utf16le':
-      case 'utf-16le':
-        return true;
-
-      default:
-        if (loweredCase)
-          return false;
-        encoding = ('' + encoding).toLowerCase();
-        loweredCase = true;
-    }
-  }
+Buffer[Symbol.for('isEncoding')] = Buffer.isEncoding = function(encoding) {
```

Comment:
I'd say either
1) Prefix the Symbol to prevent collisions (e.g. `node.isEncoding`)
2) Create and export this symbol from `internal/util.js` so it's no longer global
I lean towards the later.